Add queue management UI buttons for Edge workers

dheerajturaga opened the pull request:
- Add AddQueueButton and RemoveQueueButton components with LuListPlus/LuListMinus icons
- Create UI proxy endpoints for queue add/remove operations in worker_api/routes/ui.py
- Add QueueUpdateRequest data model for queue operations
- Update OpenAPI spec and regenerate TypeScript client code
- Integrate buttons into WorkerOperations for idle/running workers
- Include proper error handling, validation, and toast notifications
- Connect to existing add_worker_queues/remove_worker_queues functions

<img width="1907" height="745" alt="image" src="https://github.com/user-attachments/assets/8a449c5d-4999-4f48-a70b-1bdc2051ffdb" />

<img width="1918" height="657" alt="image" src="https://github.com/user-attachments/assets/417466f7-11ad-4fbd-8add-5db3ced798df" />

<img width="1918" height="661" alt="image" src="https://github.com/user-attachments/assets/61937754-cc14-44f5-baca-656a54a8bd48" />

cc: @jscheffl

jscheffl commented on providers/edge3/src/airflow/providers/edge3/worker_api/routes/ui.py:
```diff
@@ -208,3 +243,51 @@ def delete_worker(
+
+
+@ui_router.post(
+    "/worker/{worker_name}/queues/add",
```
Can you leverage the REST call options, I'd propose to PUT
```suggestion
@ui_router.put(
    "/worker/{worker_name}/queues",
```

dheerajturaga replied:
Done!

jscheffl commented on providers/edge3/src/airflow/providers/edge3/worker_api/routes/ui.py:
```diff
@@ -208,3 +243,51 @@ def delete_worker(
+
+
+@ui_router.post(
+    "/worker/{worker_name}/queues/remove",
```
Same here.
```suggestion
@ui_router.delete(
    "/worker/{worker_name}/queues",
```

jscheffl commented on providers/edge3/src/airflow/providers/edge3/plugins/www/src/components/RemoveQueueButton.tsx:
```diff
@@ -0,0 +1,156 @@
+        title="Remove Queue"
+        colorPalette="danger"
+        disabled={isDisabled}
+      >
```
Can you just "hide" the button (return `null`) instead of disabling? e.g. like:
```suggestion
  const isDisabled = availableQueues.length === 0;

  if (isDisabled) {
    return null; // nothing to delete if there is no queue entry
  }

  return (
    <>
      <IconButton
        size="sm"
        variant="ghost"
        onClick={onOpen}
        aria-label="Remove Queue"
        title="Remove Queue"
        colorPalette="danger"
      >
```

dheerajturaga replied:
Done! 

<img width="1916" height="652" alt="image" src="https://github.com/user-attachments/assets/ba59ca19-ec88-4105-8db9-92ed92f409ee" />

jscheffl reviewed (commented):
I am still sorry about the problems that compiled code needs to be constantly updated because of merge conflucts. I hope I can improve ASAP. So a rebase is needed :-(

Just some small nit and a request to change services - otherwise: COOL!

dheerajturaga committed:
Add queue management UI buttons for Edge workers

  - Add AddQueueButton and RemoveQueueButton components with LuListPlus/LuListMinus icons
  - Create UI proxy endpoints for queue add/remove operations in worker_api/routes/ui.py
  - Add QueueUpdateRequest data model for queue operations
  - Update OpenAPI spec and regenerate TypeScript client code
  - Integrate buttons into WorkerOperations for idle/running workers
  - Include proper error handling, validation, and toast notifications
  - Connect to existing add_worker_queues/remove_worker_queues functions

jscheffl commented on providers/edge3/src/airflow/providers/edge3/plugins/www/src/components/RemoveQueueButton.tsx:
```diff
@@ -0,0 +1,156 @@
+                    loadingText="Removing queue..."
+                    disabled={!selectedQueue}
+                  >
+                    <LuListMinus style={{ marginRight: "8px" }} />
```
When I compare with the <DeleteDag /> there is also no extra margin - can we make it the same here?
```suggestion
                    <LuListMinus />
```

dheerajturaga replied:
Done!

jscheffl commented on providers/edge3/src/airflow/providers/edge3/plugins/www/src/components/AddQueueButton.tsx:
```diff
@@ -0,0 +1,137 @@
+                  loading={addQueueMutation.isPending}
+                  loadingText="Adding queue..."
+                >
+                  <LuListPlus style={{ marginRight: "8px" }} />
```
When I compare with the <DeleteDag /> there is also no extra margin - can we make it the same here?
```suggestion
                  <LuListPlus />
```

dheerajturaga committed:
Jens Suggestions

jscheffl commented on providers/edge3/src/airflow/providers/edge3/plugins/www/src/components/AddQueueButton.tsx:
```diff
@@ -0,0 +1,137 @@
+                </Dialog.ActionTrigger>
+                <Button
+                  onClick={handleAddQueue}
+                  colorPalette="success"
```
Oh found a small nit at final testing: The adding button is enabled also if the text is empty. Can you disable until the user entered a queue?
```suggestion
                  colorPalette="success"
                  disabled={!queueName.trim()}
```

dheerajturaga replied:
Done!
